doc/languages-frameworks/go.section.md

```diff
@@ -62,6 +62,56 @@ The following is an example expression using `buildGoModule`:
 }
 ```
 
+### Obtaining and overriding `vendorHash` for `buildGoModule` {#buildGoModule-vendorHash}
+
+`nix-prefetch` can be used to obtain the actual hash. The following command gets the value of `vendorHash` for package `pet`:
+
+```sh
+cd path/to/nixpkgs
+nix-prefetch -E "{ hash }: ((import ./. { }).my-package.overrideAttrs (_: { vendorHash = hash; })).go-modules"
+```
+
+To obtain the hash without external tools, set `vendorHash = lib.fakeHash;` and run the build. ([more details here](#sec-source-hashes)).
+
+`vendorHash` can be overridden with `overrideAttrs`. Override the above example like this:
+
+```nix
+{
+  pet_0_4_0 = pet.overrideAttrs (finalAttrs: previousAttrs: {
+    version = "0.4.0";
+    src = fetchFromGitHub {
+      inherit (previousAttrs.src) owner repo;
+      rev = "v${finalAttrs.version}";
+      hash = "sha256-gVTpzmXekQxGMucDKskGi+e+34nJwwsXwvQTjRO6Gdg=";
+    };
+    vendorHash = "sha256-dUvp7FEW09V0xMuhewPGw3TuAic/sD7xyXEYviZ2Ivs=";
+  });
+}
+```
+
+### Overriding `goModules` (#buildGoModule-goModules-override)
+
+In the past, overriding `<pkg>.goModules` was virtually impossible (see [#86349](https://github.com/NixOS/nixpkgs/issues/86349)). Now, it is possible to override its `outputHash` and unwrapping/patching attributes by overriding `vendorHash` and corresponding attributes in the primary derivation.
+
+Overriding `<pkg>.goModules` by calling `goModules.overrideAttrs` is still unsupported. This is because when the final state of attribute `vendorHash` becomes `null`, `goModules` becomes `""`. Alternatively, the primary derivation provides an overridable `passthru.overrideModAttrs` to store the attribute overlay implicitly taken by `goModules.overrideAttrs`.
+
+```nix
+let
+  pet-foo = pkgs.overrideAttrs (finalAttrs: previousAttrs: {
+      passthru = previousAttrs.passthru // {
+        overrideModAttrs = lib.composeExtensions previousAttrs.passthru.overrideModAttrs (
+          finalModAttrs: previousModAttrs: {
+            # goModules-specific overriding goes here
```

Comment:
Do you have an idea what would someone want to do here? Perhaps [using `replace`](https://go.dev/ref/mod#go-mod-file-replace)?

Reply:
> Do you have an idea what someone would want to do here?

By searching `overrideModAttrs` across the source tree, I found the following usages:

- Patch the fetched modules using `postBuild` dynamically.
- Disable or shadow the stages inherited from the primary derivation (e.g., `preBuild = ""` or `postPatch = ""`)
- Reimplement `buildPhase` to use the vendoring commands not yet supported by `buildGoModule` (e.g., `go work vendor`).

> Perhaps [using `replace`](https://go.dev/ref/mod#go-mod-file-replace)?

If this change of the `go.mod` also fits into the `postPatch` of the primary derivation, I would prefer placing that in the primary derivation instead. I saw some of such `go.mod` patching in the codebase. A tree-wide fix may help. I need to familiarize myself with Go vendoring. Please correct me if I'm wrong.